src/push-notifications/use-push-notifications.tsx

```diff
@@ -0,0 +1,130 @@
+import {useCallback, useState} from 'react';
+import {useRegister} from '@atb/push-notifications/use-register';
+import {useConfig} from '@atb/push-notifications/use-config';
+import {PermissionsAndroid, Platform} from 'react-native';
+import messaging, {
+  FirebaseMessagingTypes,
+} from '@react-native-firebase/messaging';
+import {usePreferences} from '@atb/preferences';
+
+type PermissionStatus = 'granted' | 'denied' | 'undetermined';
+
+export const usePushNotifications = () => {
+  const {
+    preferences: {language = 'nb'},
+  } = usePreferences();
+  const [permissionStatus, setPermissionStatus] =
+    useState<PermissionStatus>('undetermined');
+  const [isLoadingPermissionStatus, setIsLoadingPermissionStatus] =
+    useState<boolean>();
+  const [isError, setIsError] = useState(false);
+  const {mutation: registerMutation} = useRegister();
+  const {query: configQuery, mutation: configMutation} = useConfig();
+
+  const checkPermissions = useCallback(() => {
+    setIsLoadingPermissionStatus(true);
+    if (Platform.OS === 'ios') {
+      messaging()
+        .hasPermission()
+        .then((status) => setPermissionStatus(mapIosPermissionStatus(status)))
+        .catch((e) => {
+          setIsError(true);
+          console.error(e);
+        })
+        .finally(() => setIsLoadingPermissionStatus(false));
+    } else if (Platform.OS === 'android') {
+      const platformVersion = Platform.Version; // Extracted outside promise to make typescript understand it's a number since Platform === 'android'
+      PermissionsAndroid.check(
+        PermissionsAndroid.PERMISSIONS.POST_NOTIFICATIONS,
+      )
+        .then((hasPermission) => {
+          // On SDK versions < 33 permission is not required, yet check() will return 'false' for those devices.
+          // However, getToken() will yield a device token for sdk < 33 even if check() returns 'false'.
+          setPermissionStatus(
+            hasPermission || platformVersion < 33 ? 'granted' : 'denied',
+          );
+        })
+        .catch((e) => {
+          setIsError(true);
+          console.error(e);
+        })
+        .finally(() => setIsLoadingPermissionStatus(false));
+    } else {
+      setIsLoadingPermissionStatus(false);
+      setIsError(true);
+    }
+  }, [Platform, messaging, PermissionsAndroid]);
+
+  const register = async () => {
```

Comment:
Wrap with `useCallback`? I see it is not a problem now as it is not used in a hook any other place, but better to be safe than sorry, just in case someone later does something like this:
```ts
useEffect(() => {
  register();
}, [register])
```